src/tree/v2/CompatibleBranchDataItem.ts

```diff
@@ -0,0 +1,57 @@
+import { AzExtParentTreeItem, AzExtTreeItem, CompatibleContextValueFilterableTreeNode, CompatibleQuickPickOptions } from '@microsoft/vscode-azext-utils';
+import * as vscode from 'vscode';
+import { ApplicationResource, Box, BranchDataProvider, ResourceModelBase } from '../../api/v2/v2AzureResourcesApi';
+import { BranchDataItemOptions, createBranchDataItemFactory } from './BranchDataItem';
+import { ResourceGroupsItem } from './ResourceGroupsItem';
+import { ResourceGroupsItemCache } from './ResourceGroupsItemCache';
+
+export class CompatibleBranchDataItem implements ResourceGroupsItem, Box, CompatibleContextValueFilterableTreeNode {
+    constructor(
+        private readonly branchItem: ResourceModelBase,
+        private readonly branchDataProvider: BranchDataProvider<ApplicationResource, ResourceModelBase>,
+        private readonly itemCache: ResourceGroupsItemCache,
+        private readonly options: BranchDataItemOptions | undefined
+    ) {
+        itemCache.addBranchItem(this.branchItem, this);
+    }
+
+    async getChildren(): Promise<ResourceGroupsItem[] | undefined> {
+        const children = await this.branchDataProvider.getChildren(this.branchItem);
+        const factory = createBranchDataItemFactory(this.itemCache);
+        return children?.map(child => factory(child, this.branchDataProvider));
+    }
+
+    public get quickPickOptions(): CompatibleQuickPickOptions {
+        const ti = this.branchItem as AzExtTreeItem;
+
+        const maybeParent = ti as AzExtParentTreeItem;
```

Comment:
As long as we're opening up the `isAzExtTreeItem` method, I think it would make sense to have a corresponding `isAzExtParentTreeItem` method to go with it. There's one implemented but it's currently internal-only.

Reply:
Created https://github.com/microsoft/vscode-azuretools/pull/1236